app/containers/GamePage/index.js

```diff
@@ -75,13 +94,62 @@ export default class GamePage extends Component {
     });
   }
 
+  loadParticipants() {
+    const { database, participants } = this.state;
+    const { params } = this.props;
+    const usersRef = database.ref(`${params.rid}/participants`);
+    usersRef.off();
+    usersRef.limitToLast(12).on('child_added', (data) => {
+      const val = data.val();
+      participants.push({ key: data.key, name: val.name, role: '', health: 5, hand: [] });
+      this.setState({ participants });
+      usersRef.once('value', (snapshot) => {
+        const num = snapshot.numChildren();
+        this.setState({ players: num });
+        if (num >= 3) {
+          this.setState({ assign: true });
+        } else {
+          this.setState({ assign: false });
+        }
+      });
+    });
+    usersRef.limitToLast(12).on('child_removed', (data) => {
+      for (let i = 0; i < participants.length; i++) {
+        if (participants[i].key === data.key) {
+          participants.splice(i, 1);
+          const num = participants.length;
+          this.setState({ participants, players: num });
+          if (num >= 3) {
+            this.setState({ assign: true });
+          } else {
+            this.setState({ assign: false });
+          }
```

Comment:
This is better than the child_added. Only thing I want to mention is you can just do this.setState({ participants, players: num, assign: num >= 3 })